src/core/custom-elements/rs-changelog.js

```diff
@@ -0,0 +1,94 @@
+// @ts-check
+/**
+ * Fetches commits since a given commitish (tag or commit) and shows them as a
+ * list of "changes since" given commitish. If a filter function is provided by
+ * the user, it is used to filter the commits that are to be shown. Otherwise,
+ * all commits are shown.
+ *
+ * @typedef {{message: string, hash: string}} Commit
+ */
+import { github } from "../github.js";
+import { hyperHTML } from "../import-maps.js";
+import { pub } from "../pubsubhub.js";
+
+let readyResolver;
+const readyPromise = new Promise(resolve => {
+  readyResolver = resolve;
+});
+
+export const name = "rs-changelog";
+
+export default class Changelog extends HTMLElement {
+  constructor() {
+    super();
+    this.props = {
+      from: this.getAttribute("from"),
+      to: this.getAttribute("to") || "HEAD",
+      /** @type {(commit: Commit) => boolean} */
+      filter:
+        typeof window[this.getAttribute("filter")] === "function"
+          ? window[this.getAttribute("filter")]
+          : () => true,
+    };
+  }
+
+  get ready() {
```

Comment:
Should this be static?

Reply:
Static won't be able to tell ready for each instance of component. Will be okay if we want each component to be used only once.

Reply:
Gave it another thought during nap. We're using the same promise, so it's essentially bugged. 🐛

Reply:
Why I wondered if this should be static :)

Reply:
We could just make each element have its own `.ready` (we should really call this `.done`) property. Then we can make `render()` private and we should be able to query for the elements, then `await` for each to be `.done`.